server/src/test/java/org/opensearch/action/support/clustermanager/TransportClusterManagerTermCheckTests.java

```diff
@@ -0,0 +1,320 @@
+        assertThat(transport.capturedRequests().length, equalTo(1));
+        CapturingTransport.CapturedRequest capturedRequest = transport.capturedRequests()[0];
+        assertTrue(capturedRequest.node.isClusterManagerNode());
+        assertThat(capturedRequest.action, equalTo("cluster:monitor/term"));
+        GetTermVersionResponse response = new GetTermVersionResponse(
+            new ClusterStateTermVersion(
+                clusterService.state().getClusterName(),
+                clusterService.state().metadata().clusterUUID(),
+                clusterService.state().term(),
+                clusterService.state().version()
+            )
+        );
+        transport.handleResponse(capturedRequest.requestId, response);
+        assertTrue(listener.isDone());
+    }
+
+    public void testTermCheckNoMatchWithClusterManager() throws ExecutionException, InterruptedException {
+        setUpCluster(Version.CURRENT);
+        TransportClusterManagerTermCheckTests.Request request = new TransportClusterManagerTermCheckTests.Request();
+
+        PlainActionFuture<TransportClusterManagerTermCheckTests.Response> listener = new PlainActionFuture<>();
+        new TransportClusterManagerTermCheckTests.Action("internal:testAction", transportService, clusterService, threadPool).execute(
+            request,
+            listener
+        );
+
+        assertThat(transport.capturedRequests().length, equalTo(1));
+        CapturingTransport.CapturedRequest termCheckRequest = transport.capturedRequests()[0];
+        assertTrue(termCheckRequest.node.isClusterManagerNode());
+        assertThat(termCheckRequest.action, equalTo("cluster:monitor/term"));
+        GetTermVersionResponse noMatchResponse = new GetTermVersionResponse(
+            new ClusterStateTermVersion(
+                clusterService.state().getClusterName(),
+                clusterService.state().metadata().clusterUUID(),
+                clusterService.state().term(),
+                clusterService.state().version() - 1
+            )
+        );
+        transport.handleResponse(termCheckRequest.requestId, noMatchResponse);
+        assertFalse(listener.isDone());
+
+        assertThat(transport.capturedRequests().length, equalTo(2));
+        CapturingTransport.CapturedRequest capturedRequest = transport.capturedRequests()[1];
+        assertTrue(capturedRequest.node.isClusterManagerNode());
+        assertThat(capturedRequest.request, equalTo(request));
+        assertThat(capturedRequest.action, equalTo("internal:testAction"));
+
+        TransportClusterManagerTermCheckTests.Response response = new TransportClusterManagerTermCheckTests.Response();
+        transport.handleResponse(capturedRequest.requestId, response);
+        assertTrue(listener.isDone());
+        assertThat(listener.get(), equalTo(response));
+
+    }
+
+    public void testTermCheckOnOldVersionClusterManager() throws ExecutionException, InterruptedException {
+
+        setUpCluster(Version.V_2_13_0);
```

Comment:
make it 2.12 as you would need to update this test post merge in 2.x which is tracking 2.13

Reply:
for older version this tracsport action is not supported right?

Reply:
Changed to 2.12. Yes it is not supported for old version and have added check [here](https://github.com/rajiv-kv/OpenSearch/blob/6ab003234999e37dcc2dc640287c1d360e46fc69/server/src/main/java/org/opensearch/action/support/clustermanager/TransportClusterManagerNodeAction.java#L277)